R/spatiotemporal.R

```diff
@@ -0,0 +1,222 @@
+                         plot = TRUE) {
+  stopifnot(
+    "`incidence_object` must have incidence class" =
+      (inherits(incidence_object, "incidence")),
+    "`incidence_object` must account for a single cumulative observation" =
+      (length(incidence_object$dates) == 1),
+    "`level` must be 0, 1, or 2" =
+      (level %in% c(0, 1, 2)),
+    "`scale` must be numeric" = (is.numeric(scale)),
+    "`threshold` must be numeric" = (is.numeric(threshold)),
+    "`plot` must be boolean" = (is.logical(plot))
+  )
+  incidence_rate <- incidence_rate(
+    incidence_object = incidence_object,
+    level = level, scale = scale
+  )
+  path_1 <- system.file("extdata", "divipola_table.rda", package = "epiCo")
+  load(path_1)
+  divipola_table <- divipola_table
+  # Match with DIVIPOLA order
+  incidence_rate_ordered <- incidence_rate[, order(match(
+    colnames(incidence_rate$counts), divipola_table$COD_MPIO
+  ))]
+  mpios <- colnames(incidence_rate_ordered$counts)
+
+  # Logarithmic transformation
+  incidence_rate_log <- log(incidence_rate_ordered$rates)
+  mpios_filtered <- as.numeric(mpios[which(incidence_rate_log > -Inf)])
+  incidence_rate_log <- incidence_rate_log[which(incidence_rate_log > -Inf)]
+
+  # Neighborhood structure
+  nb <- neighborhoods(mpios_filtered, threshold)
+  weights <- spdep::nb2listw(nb, style = "W")
+
+  # Moran's I
+  morans_i <- spdep::moran.plot(incidence_rate_log,
+    listw = weights, plot = FALSE
+  )
+  moran_data_frame <- as.data.frame(morans_i)
+  # Clusters
+  moran_data_frame <- dplyr::mutate(moran_data_frame,
+    cluster = dplyr::case_when(
+      x > mean(x) & wx >
+        mean(wx) ~ "High-High",
+      x < mean(x) & wx <
+        mean(wx) ~ "Low-Low",
+      x > mean(x) & wx <
+        mean(wx) ~ "High-Low",
+      x < mean(x) & wx >
+        mean(wx) ~ "Low-High"
+    )
+  )
+  infl_mpios <- moran_data_frame[which(moran_data_frame$is_inf), ]
+  morans_index <- c(
+    list(municipios = moran_data_frame$labels),
+    list(quadrant = moran_data_frame$cluster),
+    list(influential = moran_data_frame$is_inf),
+    list(logIncidence = moran_data_frame$x),
+    list(lagIncidence = moran_data_frame$wx)
+  )
+  if (!all(is.na(morans_index$quadrant))) {
+    cat(paste("Influential municipalities are:", "\n"))
+    # Influential observations
+    for (i in seq_len(nrow(infl_mpios))) {
+      if (!is.na(infl_mpios$cluster[i])) {
+        # nolint start: string_boundary_linter
+        relative_incidence <- ifelse(substr(
+          infl_mpios$cluster[i], 1, 1
+        ) == "H", "high", "low")
+        # nolint end: string_boundary_linter
+        relative_correlation <- ifelse(substr(
+          infl_mpios$cluster[i], 2, 2
+        ) == "H", "high", "low")
+        cat(paste(
+          infl_mpios$labels[i], "with", relative_incidence,
+          "incidence and", relative_correlation, "spatial correlation",
+          "\n"
+        ))
+      }
+    }
```

Comment:
We can get rid of the loop entirely with this simpler and faster solution:

```suggestion
    cat(
      sprintf(
        "%s with %s incidence and %s spatial correlation",
        infl_mpios$labels,
        ifelse(substr(infl_mpios$cluster, 1, 1) == "H", "high", "low"),
        ifelse(substr(infl_mpios$cluster, 2, 2) == "H", "high", "low")
      ), 
    sep = "\n")
```